File: server/src/main/java/de/davelee/trams/server/request/AddTimeRequest.java
package de.davelee.trams.server.request;

/**
 * This class is part of the TraMS Server REST API. It represents a request to add the number of minutes
 * to the time for a particular company.
 * @author Dave Lee
 */
public class AddTimeRequest {

    /**
     * The name of the company to add the time to.
     */
    private String company;

    /**
     * The number of minutes to add to the time.
     */
    private int minutes;

    public AddTimeRequest() {
    }

    public AddTimeRequest(String company, int minutes) {
        this.company = company;
        this.minutes = minutes;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public int getMinutes() {
        return minutes;
    }

    public void setMinutes(int minutes) {
        this.minutes = minutes;
    }

    @Override
    public String toString() {
        return "AddTimeRequest{" +
                "company='" + company + '\'' +
                ", minutes=" + minutes +
                '}';
    }
}

File: desktop/src/app/livesituation/addtime.request.ts
/**
 * This class defines a model for requests to add time in minutes in simulation mode in TraMS.
 */
export class AddTimeRequest {

    private company: string;
    private minutes: number;

    /**
     * Construct a new request to add time in minutes.
     * @param company the company that we should add time for.
     * @param minutes the number of minutes that we should add to the time.
     */
    constructor(company: string, minutes: number) {
        this.company = company;
        this.minutes = minutes;
    }
}